source/common/http/conn_manager_impl.cc

```diff
@@ -1544,8 +1602,13 @@ void ConnectionManagerImpl::ActiveStreamFilterBase::commonContinue() {
 
   ENVOY_STREAM_LOG(trace, "continuing filter chain: filter={}", parent_,
                    static_cast<const void*>(this));
-  ASSERT(stopped_);
-  stopped_ = false;
+  ASSERT(!canIterate());
+  if (iteration_state_ == IterationState::StopSingleIteration) {
+    // Only allow iteration if it was stopped for one frame type. If the iteration has stopped for
```

Comment:
I thought we split the iteration state into "what should the state be" and "what filter do we iterate from" to avoid this problem.  Shouldn't we now be able to consistently set allowIteration at the top here, and consistently set iterate_from_current_filter at the bottom, or am I misremembering how this works?

Reply:
Good point! Fixed.